hashes/src/hash160.rs

```diff
@@ -89,8 +87,8 @@ mod tests {
             // Hash through high-level API, check hex encoding/decoding
             let hash = hash160::Hash::hash(&test.input[..]);
             assert_eq!(hash, test.output_str.parse::<hash160::Hash>().expect("parse hex"));
-            assert_eq!(&hash[..], &test.output[..]);
```

Comment:
Creating a slice like this is a hang over from when we used and old Rust version as our MSRV (Rust 1.29 I think). Could you put these changes in a separate patch please.